src/riak_kv_entropy_manager.erl

```diff
@@ -219,6 +221,20 @@ clear_trees() ->
 get_version() ->
     gen_server:call(?MODULE, get_version, infinity).
 
+-spec get_partition_version(index()) -> version().
+get_partition_version(Index) ->
+    try
+        case ets:lookup(?ETS, Index) of
+            [] ->
+                legacy;
+            [{Index, Version}] ->
+                Version
+        end
+    catch
```

Comment:
Why the try/catch here? In case you somehow end up with multiple rows in the ETS table? ets:lookup docs seem to indicate that it should never throw, and given the table is a set (can only have one entry per key) it'd never actually return more than one row. Was there something else you were defending against here?